plugins/ingest-attachment/src/main/java/org/elasticsearch/ingest/attachment/AttachmentProcessor.java

```diff
@@ -0,0 +1,202 @@
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.ingest.attachment;
+
+import org.apache.tika.language.LanguageIdentifier;
+import org.apache.tika.metadata.Metadata;
+import org.apache.tika.metadata.TikaCoreProperties;
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.common.Base64;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.ingest.core.AbstractProcessor;
+import org.elasticsearch.ingest.core.AbstractProcessorFactory;
+import org.elasticsearch.ingest.core.IngestDocument;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.elasticsearch.ingest.core.ConfigurationUtils.newConfigurationException;
+import static org.elasticsearch.ingest.core.ConfigurationUtils.readIntProperty;
+import static org.elasticsearch.ingest.core.ConfigurationUtils.readOptionalList;
+import static org.elasticsearch.ingest.core.ConfigurationUtils.readStringProperty;
+
+public final class AttachmentProcessor extends AbstractProcessor {
+
+    public static final String TYPE = "attachment";
+
+    private static final int NUMBER_OF_CHARS_INDEXED = Integer.MAX_VALUE;
+
+    private final String sourceField;
+    private final String targetField;
+    private final Set<Field> fields;
+    private final int indexedChars;
+
+    AttachmentProcessor(String tag, String sourceField, String targetField, Set<Field> fields,
+                        int indexedChars) throws IOException {
+        super(tag);
+        this.sourceField = sourceField;
+        this.targetField = targetField;
+        this.fields = fields;
+        this.indexedChars = indexedChars;
+    }
+
+    @Override
+    public void execute(IngestDocument ingestDocument) {
+        String base64Input = ingestDocument.getFieldValue(sourceField, String.class);
+
+        Metadata metadata = new Metadata();
+        try {
+            byte[] decodedContent = Base64.decode(base64Input.getBytes(UTF_8));
+            String parsedContent = TikaImpl.parse(decodedContent, metadata, indexedChars);
+
+            if (fields.contains(Field.CONTENT)) {
+                // somehow tika seems to append a newline at the end automatically, lets remove that again
```

Comment:
Interesting. I never wanted to remove what is returned by Tika but I agree that it makes sense.